### PR TITLE
Bitmap

### PR DESCRIPTION
**Pull request by Theodus**
I figured that the bitmapping used in the Dtrie could also be useful to other data structures. I will change the Dtrie implementation to use this package but I was not sure if you would want that done in a separate PR.

Benchmarks for PopCount:
BenchmarkPopCount32-4	1000000000	         2.09 ns/op
BenchmarkPopCount64-4	1000000000	         2.59 ns/op

### CODE REVIEW

**Comment by alexandercampbell-wf** [comment]
> I will change the Dtrie implementation to use this package but I was not sure if you would want that done in a separate PR.

I'd prefer it was done as part of the same PR; the code usually ends up cleaner if there's at least one consumer from the outset.

I'd almost prefer this was part of the `bitarray` package. Not necessarily as an implementation of the `bitarray.BitArray` interface, but maybe as a separate, exported struct labelled `BitArray32`. Alternatively, this bitmap package could become a subdirectory underneath `bitarray`. Thoughts?

FYI @dustinhiatt-wf @tylertreat-wf

**Comment by dustinhiatt-wf** [comment]
Yeah, I feel as if this should be a part of that package.  Technically, you can create a new bit array of size 32 or 64 and you'll get the same functionality but at a higher cost (still O(1) but more overhead).

**Comment by Theodus** [comment]
That's fine with me.

**Comment by Theodus** [comment]
The next commit is the last one. It will fail because the import path of bitarray is changed to the Workiva version and it has to be merged before it will compile.

**Comment by Theodus** [comment]
Disregard the last comment, I underestimated Travis CI.

File: bitarray/bitmap.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	return b | (1 << pos)
+}
+
+// ClearBit returns a Bitmap32 with the bit at the given position set to 0
+func (b Bitmap32) ClearBit(pos uint) Bitmap32 {
+	return b & ^(1 << pos)
+}
+
+// HasBit returns true if the bit at the given position in the Bitmap32 is 1
+func (b Bitmap32) HasBit(pos uint) bool {
```
Comment (alexandercampbell-wf) [inline bitarray/bitmap.go]: We use the name `GetBit` elsewhere in the code to be consistent with our convention of `SetBit`. Mind changing it here as well?

**Comment by Theodus** [comment]
No problem.

**Comment by alexandercampbell-wf** [comment]
+1
@dustinhiatt-wf @tylertreat-wf

**Comment by dustinhiatt-wf** [comment]
+1

**Pull request merged**